sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -109,7 +109,6 @@ std::vector<EventImplPtr> Scheduler::getWaitList(EventImplPtr Event) {
 }
 
 void Scheduler::waitForEvent(EventImplPtr Event) {
-  std::lock_guard<std::mutex> lock(MGraphLock);
   GraphProcessor::waitForEvent(std::move(Event));
```

Comment:
Is `GraphProcessor::waitForEvent` mothed thread safe?
If not, there is another solution: split critical section and use a couple of mutexes.

Reply:
It must be thread safe.